core/templates/dev/head/components/HintAndSolutionButtonsDirective.ts

```diff
@@ -21,8 +21,10 @@ oppia.directive('hintAndSolutionButtons', [
     return {
       restrict: 'E',
       scope: {},
+      bindToController: true,
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/components/hint_and_solution_buttons_directive.html'),
+      controllerAs: 'hintAndSolutionButtonsCtrl',
```

Comment:
Just to check, any reason we don't go for $ctrl like in https://angular.io/guide/upgrade#using-component-directives ?

Reply:
No specific reason, I just thought it would be better to name each controller according to the directive. I am fine with both of the ways. Although now I feel $ctrl will be better to have consistency everywhere in all templates. Shall I make the changes for the same?

Reply:
That's your call :)

Reply:
(But please say which one you decide on, and why. If you are not sure, I would advise following standards where possible.)

Reply:
I have decided to go with $ctrl since it is more consistent to use the same controllerAs syntax everywhere as mentioned in standards rather than having a new name for each of the controller.